src/app/contributors/store/contributors.store.ts

```diff
@@ -0,0 +1,44 @@
+import { HttpErrorResponse } from '@angular/common/http';
+import { inject, Injectable } from '@angular/core';
+import { ComponentStore, tapResponse } from '@ngrx/component-store';
+import { pipe, switchMap, tap } from 'rxjs';
+import { GithubService } from 'src/app/services/github.service';
+import { ContributorsModel } from '../contributors.model';
+
+interface ContributorsStateModel {
+  contributors: ContributorsModel[];
+  isLoading: boolean;
+  error?: Error | null;
+}
+
+@Injectable()
+export class ContributorsStore extends ComponentStore<ContributorsStateModel> {
+  private githubService = inject(GithubService);
+  constructor() {
+    super({
+      contributors: [],
+      isLoading: false,
+    });
+  }
+
+  readonly vm = this.selectSignal(({ contributors, isLoading }) => ({
+    contributors,
+    isLoading,
+  }));
+
+  readonly getContributors = this.effect<void>(
```

Comment:
suggestion/nit: if `getContributors` is only called in the component's `ngOnInit`, we could use `OnStateInit` hook ( it is a `ComponentStore`'s life cycle hook) instead of having to call it in the component. I don't have strong opinion about this though so it's your call. I personally like my component as clean as possible